nullaway/src/main/java/com/uber/nullaway/fixserialization/FixSerializationConfig.java

```diff
@@ -0,0 +1,180 @@
+  }
+
+  public FixSerializationConfig(
+      boolean suggestEnabled,
+      boolean suggestEnclosing,
+      AnnotationConfig annotationConfig,
+      String outputDirectory) {
+    this.suggestEnabled = suggestEnabled;
+    this.suggestEnclosing = suggestEnclosing;
+    this.outputDirectory = outputDirectory;
+    this.annotationConfig = annotationConfig;
+    serializer = new Serializer(this);
+  }
+
+  /**
+   * Sets all flags based on their values in the configuration file.
+   *
+   * @param configFilePath Path to the serialization config file written in xml.
+   */
+  public FixSerializationConfig(String configFilePath) {
+    Preconditions.checkNotNull(configFilePath);
+    Document document;
+    try {
+      DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder builder = factory.newDocumentBuilder();
+      document = builder.parse(Files.newInputStream(Paths.get(configFilePath)));
+      document.normalize();
+    } catch (Exception e) {
+      throw new RuntimeException(
+          "Error in reading/parsing config at path: " + configFilePath + "\n" + e);
+    }
+    this.outputDirectory =
+        XMLUtil.getValueFromTag(document, "serialization:path", String.class).orElse(null);
+    Preconditions.checkNotNull(
+        this.outputDirectory, "Error in FixSerialization Config: Output path cannot be null");
+    suggestEnabled =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "active", Boolean.class)
+            .orElse(false);
+    suggestEnclosing =
+        XMLUtil.getValueFromAttribute(document, "serialization:suggest", "enclosing", Boolean.class)
+                .orElse(false)
+            && suggestEnabled;
+    String nullableAnnot =
+        XMLUtil.getValueFromTag(document, "serialization:annotation:nullable", String.class)
+            .orElse("javax.annotation.Nullable");
+    String nonnullAnnot =
+        XMLUtil.getValueFromTag(document, "serialization:annotation:nonnull", String.class)
+            .orElse("javax.annotation.Nonnull");
+    this.annotationConfig = new AnnotationConfig(nullableAnnot, nonnullAnnot);
+    serializer = new Serializer(this);
+  }
+
+  /**
+   * Validates whether {@code suggestEnabled} is active and the element involved in the error is
+   * accessible and not received in bytecodes.
+   *
+   * @param trees Trees object.
+   * @param symbol Symbol of the targeted element to perform type change.
+   * @return true, if the element is accessible and {@code suggestEnabled} is active.
+   */
+  public boolean canFixElement(Trees trees, Element symbol) {
+    if (!suggestEnabled) return false;
+    if (symbol == null) return false;
+    return trees.getPath(symbol) != null;
+  }
+
+  /** Builder class for Serialization Config */
+  public static class Builder {
+
+    private boolean suggestEnabled;
+    private boolean suggestEnclosing;
+    private String nullable;
+    private String nonnull;
+    private String outputDir;
```

Comment:
This one is going to have to be `@Nullable` here once we are running NullAway on NullAway

Reply:
Sorry but I did not fully understand what I have to do here, would you please give some more detailed instructions. Thank you very much.

Reply:
`outputDir` is not being initialized by the constructor, and is thus a `@Nullable` field. We can worry about this as part of another ongoing PR to enroll NullAway itself into NullAway checking, though, but just as general observation, this field is `@Nullable`.